crates/valence_entity/src/active_status_effects.rs

```diff
@@ -0,0 +1,148 @@
+    effect: StatusEffect,
+    amplifier: i8,
+    /// The total duration of the status effect in ticks.
+    duration: i32,
+    ambient: bool,
+    show_particles: bool,
+    show_icon: bool,
+}
+
+impl ActiveStatusEffect {
+    /// Creates a new [`ActiveStatusEffect`].
+    pub fn from_effect(effect: StatusEffect) -> Self {
+        Self {
+            effect,
+            amplifier: 0,
+            duration: 600,
+            ambient: false,
+            show_particles: true,
+            show_icon: true,
+        }
+    }
+
+    /// Sets the amplifier of the [`ActiveStatusEffect`].
+    pub fn with_amplifier(mut self, amplifier: i8) -> Self {
+        self.amplifier = amplifier;
+        self
+    }
+
+    /// Sets the duration of the [`ActiveStatusEffect`] in ticks.
+    pub fn with_duration(mut self, duration: i32) -> Self {
+        self.duration = duration;
+        self
+    }
+
+    /// Sets the duration of the [`ActiveStatusEffect`] in seconds.
+    pub fn with_duration_seconds(mut self, duration: f32) -> Self {
+        self.duration = (duration * 20.0) as i32;
+        self
+    }
+
+    /// Sets the duration of the [`ActiveStatusEffect`] to infinite.
+    pub fn with_infinite_duration(mut self) -> Self {
+        self.duration = -1; // -1 is infinite in vanilla
+        self
+    }
+
+    /// Sets whether the [`ActiveStatusEffect`] is ambient.
+    pub fn with_ambient(mut self, ambient: bool) -> Self {
+        self.ambient = ambient;
+        self
+    }
+
+    /// Sets whether the [`ActiveStatusEffect`] shows particles.
+    pub fn with_show_particles(mut self, show_particles: bool) -> Self {
+        self.show_particles = show_particles;
+        self
+    }
+
+    /// Sets whether the [`ActiveStatusEffect`] shows an icon.
+    pub fn with_show_icon(mut self, show_icon: bool) -> Self {
+        self.show_icon = show_icon;
+        self
+    }
+
+    /// Decrements the duration of the [`ActiveStatusEffect`] by a tick.
+    pub fn decrement_duration(&mut self) {
+        if self.duration < 0 {
+            return;
+        }
+        self.duration -= 1;
+        if self.duration < 0 {
+            self.duration = 0;
+        }
+    }
+
+    /// Returns the [`StatusEffect`] of the [`ActiveStatusEffect`].
+    pub fn status_effect(&self) -> StatusEffect {
+        self.effect
+    }
+
+    /// Returns the amplifier of the [`ActiveStatusEffect`].
+    pub fn amplifier(&self) -> i8 {
+        self.amplifier
+    }
+
+    /// Returns the remaining duration of the [`ActiveStatusEffect`] in ticks.
+    pub fn duration(&self) -> i32 {
+        self.duration
+    }
+
+    /// Returns whether the [`ActiveStatusEffect`] is ambient.
+    pub fn ambient(&self) -> bool {
+        self.ambient
+    }
+
+    /// Returns whether the [`ActiveStatusEffect`] shows particles.
+    pub fn show_particles(&self) -> bool {
+        self.show_particles
+    }
+
+    /// Returns whether the [`ActiveStatusEffect`] shows an icon.
```

Comment:
NIT: For all these comments I prefer the format of: Returns [true] if the [thing] has [something]. (like how you did it with `expired`)